units/Mario.cfg

```diff
@@ -4,6 +4,9 @@
     id=Peasant_Mario
 #ifndef MULTIPLAYER
     advances_to=Spearman,Bowman,Sergeant
+#endif
+#ifdef HARD
+    advances_to=Spearman,Bowman
```

Comment:
Yes (if he doesn't advance to spearman/bowman during C1S1).  The Duke is really, really useful, to the point of being a bit of a crutch.

Reply:
I think he's way too beneficial to be allowed on hard.  Jacking up the experience would help, but you'll get him eventually.
The Duke in part 2 is worse, though.  He's still available, on easy/normal.

And there's no guarantee you get him anyway.  In my current run, he levelled to a spearman in S1. I used to think that wasn't possible, as he always ended S1 with max_exp-1, but I guess that was just dumb luck.

If it wasn't so much work, I'd almost consider making him an autorecall, so that you can't keep him safely hidden on the recall list whenever there is danger.  At least until he advances.  That also may backfire, he's one heck of a unit for a free recall.

Reply:
And from the sounds of it, he's very important when playing without items...

https://forums.wesnoth.org/viewtopic.php?p=688326#p688326

Reply:
> I think he's way too beneficial to be allowed on hard. Jacking up the experience would help, but you'll get him eventually.

If he had like 300% more exp required to advance (like an extremely exaggerated form of the _dim_ trait the goblins can have), you'd get him so much farther in the campaign that his usefulness would be comparable to part 2, no? Also, he would get AMLAs much slower, like if he had taken 6 AMLAs before. This would also make it night impossible to miss him due to advancing too early.

> It might be nice to give him one or two Mario-specific upgrades if the takes the bow/spear path, since he is special (IMO). 

How about a _perfectionist_ trait, making him advance much slower but deal 50% more damage? Based on the idea that he could be learning slowly and carefully, but becoming much stronger after finally learning it. This would make the other career paths more attractive for him, while the bonus would be rather insignificant for a Duke.

> Or give Connor a bump. I've always felt he was supposed to be special (more than just highlander) in some way, but he just doesn't measure up. I make survival of Conner, Mario, Ethiel and Skellie an unwritten objective for myself just because it should be that way IMO.

There are many other characters who enter the campaign with some role for a short time and then become insignificant. It seems like a cool idea to give them some properties.

Reply:
Okay, for now how about we go with the +300% XP on hard and then come back to the advancement paths separately?  That way I won't get too distracted (and have to start over from ch1s1, again).  I assume this needs to be done to both Mario.cfg and Duke.cfg.

Reply:
No, I meant doing this to Mario in scenario 1:

```
[modifications]
  [advancement]
    [effect]
      apply_to=max_experience
      increase=300%
    [/effect]
    [effect]
      apply_to=attack
      increase_damage=50%
    [/effect]
  [/advancement]
[/modifications]
```

Reply:
Is that going to carry through all advancement paths?  Given the choice (and it looks like this will virtually guarantee the player gets a choice), the player should always choose sarge unless there is a very strong incentive to pick something else.  Duke certainly needs to be knocked way down(out) on hard, but I'm not sure the point of borking the others.  

I'd be horrified at the idea of giving Duke extra damage on hard, except he's really not an offensive unit.

Reply:
> Is that going to carry through all advancement paths?

Yes. If he doesn't choose Sergeant, he'll have a huge damage potential. It would be somewhat less than 50%, because the exp malus would reduce the number of AMLAs he takes (he would be getting them with a frequency comparable to a unit having taken about 7.5 AMLAs), and traits modify damage before additive effects of AMLAs and items are applied.

> I'd be horrified at the idea of giving Duke extra damage on hard, except he's really not an offensive unit.

My point of that is that the bonus would be of little value on Duke because he's not an offensive unit.

Reply:
Almost there.  I think the remain question is do I wrap this in ifdef HARD?

It's almost certainly going to have a big effect.  I'm loathe to touch EASY, as I know there are those who expect easy to be a few orders of magnitude below trivial.

If it only applies to hard, then the Duke line can be unavailable on easy/normal if Mario advances in S1, but is guaranteed on hard.  Seems backward if anything.

Reply:
Maybe setting the experience malus to 100% / 200% / 300% depending on difficulty could do the trick. Even on easy, it would make him special (which I think is a good idea), with the damage bonus being probably worth the exp malus, and furthermore, he would be almost certainly advanceable to Sergeant.

>  I'm loathe to touch EASY, as I know there are those who expect easy to be a few orders of magnitude below trivial.

I remember one player asking on the forums how to get past some parts of the campaign on easy. It was just one, but I think she was the only one of this group of players to actually reach out to the forums and ask. So I agree with the premise that easy should be really easy.

But in this case, it's questionable if a player who needs to play on easy can take advantage of Mario's ability to become a crutch character.

Reply:
Good idea.

I'd lean towards also layering the damage bonus, like 50/40/25 or something, but I don't see any reason to fuss over that now.  Probably want to change it anyway if you decide to do something interesting with the advancement routes later.

Reply:
> I'd lean towards also layering the damage bonus, like 50/40/25 or something, but I don't see any reason to fuss over that now.

If he had only 25% on hard with the 300% exp malus, the choice would be between a crappy Swordmaster and actually having a Duke. It has to outweigh 7.5 AMLAs to be worth it for the Swordmaster path.